graylog2-web-interface/src/pages/ConfigurationsPage.tsx

```diff
@@ -0,0 +1,240 @@
+ * it under the terms of the Server Side Public License, version 1,
+ * as published by MongoDB, Inc.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * Server Side Public License for more details.
+ *
+ * You should have received a copy of the Server Side Public License
+ * along with this program. If not, see
+ * <http://www.mongodb.com/licensing/server-side-public-license>.
+ */
+import * as React from 'react';
+import { useEffect, useMemo, useState } from 'react';
+import { chunk } from 'lodash';
+import { ErrorBoundary } from 'react-error-boundary';
+import { SystemConfiguration } from 'views/types';
+
+import { Col, Row } from 'components/graylog';
+import { DocumentTitle, PageHeader, Spinner } from 'components/common';
+import { useStore } from 'stores/connect';
+import CombinedProvider from 'injection/CombinedProvider';
+import { isPermitted } from 'util/PermissionsMixin';
+import SearchesConfig from 'components/configurations/SearchesConfig';
+import MessageProcessorsConfig from 'components/configurations/MessageProcessorsConfig';
+import SidecarConfig from 'components/configurations/SidecarConfig';
+import EventsConfig from 'components/configurations/EventsConfig';
+import UrlWhiteListConfig from 'components/configurations/UrlWhiteListConfig';
+import {} from 'components/maps/configurations';
+import style from 'components/configurations/ConfigurationStyles.lazy.css';
+import { Store } from 'stores/StoreTypes';
+import usePluginEntities from 'views/logic/usePluginEntities';
+
+import DecoratorsConfig from '../components/configurations/DecoratorsConfig';
+
+const { CurrentUserStore } = CombinedProvider.get('CurrentUser');
+const { ConfigurationsActions, ConfigurationsStore } = CombinedProvider.get('Configurations');
+
+const SEARCHES_CLUSTER_CONFIG = 'org.graylog2.indexer.searches.SearchesClusterConfig';
+const MESSAGE_PROCESSORS_CONFIG = 'org.graylog2.messageprocessors.MessageProcessorsConfig';
+const SIDECAR_CONFIG = 'org.graylog.plugins.sidecar.system.SidecarConfiguration';
+const EVENTS_CONFIG = 'org.graylog.events.configuration.EventsConfiguration';
+const URL_WHITELIST_CONFIG = 'org.graylog2.system.urlwhitelist.UrlWhitelist';
+
+type ErrorFallbackProps = {
+  error: {
+    message: string,
+  },
+  title: string,
+};
+
+const ErrorFallback = ({ error, title }: ErrorFallbackProps) => (
+  <>
+    <h2>{title}</h2>
+    <p>Something went wrong:</p>
+    <pre>{error.message}</pre>
+  </>
+);
+
+type BoundaryProps = {
+  children: React.ReactNode,
+  title: string,
+}
+
+const Boundary = ({ children, title }: BoundaryProps) => (
+  <ErrorBoundary FallbackComponent={(props) => <ErrorFallback title={title} {...props} />}>
+    {children}
+  </ErrorBoundary>
+);
+
+const ConfigletContainer = ({ children, title }: BoundaryProps) => (
```

Comment:
I think the component is way clearer now that before but I also have the impression that the component would benefit of splitting some of its inline components into their own files, making it easier to re-use them and also reducing the amount of concepts inside the file. `ConfigletContainer` and its related components would be a good candidate, I think, since they provide a valuable component for other parts of the application rendering configuration options, and would make `ConfigurationsPage` more focused on its own purpose. What do you think?

Reply:
Great idea. I had the same impression, but was a bit hesitant to include that step in the PR. Thanks for the motivation! I extracted the `ConfigletContainer` and `PluginConfigRows` components.

Reply:
I think it looks much cleaner now, thank you! 👍